### PR TITLE
Added Inversion Count using merge sort

### PR DESCRIPTION
**Pull request by kaustubh-pandey**
Fixes # 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Sample Input / Output.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:

- Added Inversions Count using Merge Sort approach

#### Languages Used:
- C++

#### Files Added:
- Inversions_Count.cpp

### CODE REVIEW

**Comment by kaustubh-pandey** [comment]
In reference to issue #817

File: Inversions_Count/Inversions_Count.cpp [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+int main(){
+    /*
+    Count number of inversions for the input_arr [1,10,3,4,5,8,7,9,2]
+    */
+    int input_arr[]={1,10,3,4,5,8,7,9,2};
+    int n=sizeof(input_arr)/sizeof(input_arr[0]);
+    cout<<inversions_count(input_arr,n)<<endl;
+	return 0;
```
Comment (GOVINDDIXIT) [inline Inversions_Count/Inversions_Count.cpp]: fix indentation

**Comment by kaustubh-pandey** [comment]
I have fixed the indentation

**Comment by marwahmanbir** [comment]
Fix and complete the PR template above.

**Comment by kaustubh-pandey** [comment]
Done

File: Inversions_Count/Inversions_Count.cpp [hunk 2]
```diff
@@ -0,0 +1,60 @@
+#include<bits/stdc++.h>
+using namespace std;
+
+//merge two sorted arrays such that resultant array is also sorted
+int merge(int input_arr[], int *aux,int low,int mid ,int high){
+    int x=low,y=mid,z=low,count=0;
```
Comment (marwahmanbir) [inline Inversions_Count/Inversions_Count.cpp]: Add space after every comma.

File: Inversions_Count/Inversions_Count.cpp [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    while((x<=mid-1) && (y<=high)){
```
Comment (marwahmanbir) [inline Inversions_Count/Inversions_Count.cpp]: Add space before ``{``.

File: Inversions_Count/Inversions_Count.cpp [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        if(input_arr[x]<=input_arr[y]){
```
Comment (marwahmanbir) [inline Inversions_Count/Inversions_Count.cpp]: Space around ``<=`` and before ``{`` in every statement that contain these.

File: Inversions_Count/Inversions_Count.cpp [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+            aux[z++]=input_arr[x++];
```
Comment (marwahmanbir) [inline Inversions_Count/Inversions_Count.cpp]: Space around ``=`` in every statement.

File: Inversions_Count/Inversions_Count.cpp [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+        }
+        else{
+            aux[z++]=input_arr[y++];
+            count+=mid-x;
```
Comment (marwahmanbir) [inline Inversions_Count/Inversions_Count.cpp]: Space around ``+=`` and ``-``

File: Inversions_Count/Inversions_Count.cpp [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    for(int i=low;i<=high;i++){
+        input_arr[i]=aux[i];
+    }
+    return count;
+}
+
+//merge sort to find inversion count
+int mergeSort(int input_arr[],int *aux,int low,int high){
```
Comment (marwahmanbir) [inline Inversions_Count/Inversions_Count.cpp]: Space after commas.

File: Inversions_Count/Inversions_Count.cpp [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+    int count=0;
+    if(high>low){
+        int mid=(low+high)/2;
+        count = mergeSort(input_arr, aux, low, mid);
+        count += mergeSort(input_arr, aux, mid + 1, high);
+        count += merge(input_arr, aux, low, mid + 1, high);
```
Comment (marwahmanbir) [inline Inversions_Count/Inversions_Count.cpp]: The spacing is perfect in above three lines.

File: Inversions_Count/Inversions_Count.cpp [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+
+//wrapper function that returns number of inversions
+int inversions_count(int input_arr[],int n){
+    int *aux=new int[n];
+    return mergeSort(input_arr,aux,0,n-1);
+}
+
+int main(){
+    /*
+    Count number of inversions for the input_arr [1,10,3,4,5,8,7,9,2]
```
Comment (marwahmanbir) [inline Inversions_Count/Inversions_Count.cpp]: Every single line comment should be added with ``//`` and should have a single blank line before it.

File: Inversions_Count/Inversions_Count.cpp [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+    */
+    int input_arr[]={1,10,3,4,5,8,7,9,2};
```
Comment (marwahmanbir) [inline Inversions_Count/Inversions_Count.cpp]: Space after every comma

File: Inversions_Count/Inversions_Count.cpp [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+    int n=sizeof(input_arr)/sizeof(input_arr[0]);
```
Comment (marwahmanbir) [inline Inversions_Count/Inversions_Count.cpp]: Space around ``=`` and around ``/``

**Comment by marwahmanbir** [comment]
The spaces and newlines are according to coding conventions (https://www.oracle.com/technetwork/java/javase/documentation/codeconventions-141388.html).

**Comment by kaustubh-pandey** [comment]
I have made the changes. Please check

**Pull request merged**